src/scheduler.rs

```diff
@@ -65,32 +68,62 @@ async fn scheduler_loop(
                 checks_scheduled,
                 "scheduler.tick_execution_complete"
             );
+            tick
         });
+
+        tick_complete_tx
+            .send(tick_complete_join_handle)
+            .expect("Failed to queue join handle for tick completion");
     };
 
+    let in_order_tick_processor_join_handle = tokio::spawn(async move {
+        while let Some(tick_complete_join_handle) = tick_complete_rx.recv().await {
```

Comment:
What case would this be `None`? Just when we terminate things?

Is there any chance it can return `None` if there's nothing to `.recv`?

Reply:
When the channel closes it becomes none. otherwise it waits to receive more things

https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.recv

>The channel is closed when all senders have been dropped, or when [close](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.close) is called.